cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -582,6 +582,17 @@ public RunRecordDetail getRun(ProgramReference programRef, String runId) {
     });
   }
 
+  @Override
+  public void markApplicationsLatest(Collection<ApplicationId> appIds)
+      throws IOException, ApplicationNotFoundException {
+    TransactionRunners.run(transactionRunner, context -> {
+      AppMetadataStore mds = getAppMetadataStore(context);
+      for (ApplicationId appId : appIds) {
+        mds.updateLatestApplicationVersion(appId);
+      }
+    }, IOException.class, ApplicationNotFoundException.class);
```

Comment:
@GnsP I thought we want to skip missing versions and continue marking rest of them latest ?

Reply:
That was for the updateGitMeta. In case of marking latest, if any of the apps are not marked latest then the entire operation should fail. Because there may be dependencies between pipelines/apps.

Reply:
Makes sense thanks